ark/schema/structure/structure.ts

```diff
@@ -299,6 +324,40 @@ const implementation: nodeImplementationOf<Structure.Declaration> =
 
 				return childIntersectionResult
 			}
+		},
+		reduce: (inner, $) => {
+			if (inner.index) {
+				let hasUpdated = false
+
+				const requiredProps = inner.required ?? []
+				const optionalProps = inner.optional ?? []
+				const newOptionalProps: OptionalNode[] = [...optionalProps]
+
+				if (requiredProps.length === 0 && optionalProps.length === 0) return
```

Comment:
Can we instead just directly return early if neither was defined?

We don't need to worry about checking length 0 as by this point they will have been stripped if that was true.

Reply:
Adjusted in 72562be0